ledger/ledger.go

```diff
@@ -269,7 +268,25 @@ func initBlocksDB(tx *sql.Tx, l *Ledger, initBlocks []bookkeeping.Block, isArchi
 				return err
 			}
 		}
+
+		// Manually replace block 0, even if we already had it
+		// (necessary to normalize the payset commitment because of a
+		// bug that caused its value to change)
+		//
+		// Don't bother for non-archival nodes since they will toss
+		// block 0 almost immediately
+		//
+		// TODO remove this once a version containing this code has
+		// been deployed to archival nodes
+		if len(initBlocks) > 0 && initBlocks[0].Round() == basics.Round(0) {
```

Comment:
Ideally we wouldn't have this piece of code being executed upon every startup, however, given that we're going to remove this after two consecutive consensus upgrade releases, I think it can stay as is.